### PR TITLE
Nginx: added OpenSSL for local development

### PR DESCRIPTION
**Pull request by Shchepotin**
##### I completed the 3 steps below:

- I've read the [Contribution Guide](http://laradock.io/contributing).
- I've updated the **documentation**. (refer to [this](http://laradock.io/contributing/#update-the-documentation-site) for how to do so).
- I enjoyed my time contributing and making developer's life easier :)

### CODE REVIEW

**Comment by jasperf** [comment]
Do we not need a volume to store the certificates as well?

**Comment by Shchepotin** [comment]
It makes sense.

File: nginx/Dockerfile [hunk 1]
```diff
@@ -4,17 +4,6 @@ LABEL maintainer="Mahmoud Zalt <mahmoud@zalt.me>"
 
 COPY nginx.conf /etc/nginx/
 
-# Install OpenSSL
-
-ARG NGINX_INSTALL_SSL=${NGINX_INSTALL_SSL}
-RUN if [ ${NGINX_INSTALL_SSL} = true ]; then \
-    apk add --no-cache openssl \
```
Comment (jasperf) [inline nginx/Dockerfile]: We may not need auto generation, but we still need openssl installed don't we?

**Comment by jasperf** [comment]
Well, looks good to me though I am not sure whether we should add an ssl certificate. Anyways, we will have to wait for @Mahmoudz to find time to check this pull request now. I'd love to have SSL capability added. Was missing it and was considering using `ADD` or a volume. Volume seems to be a nicer way.

**Comment by jasperf** [comment]
also considered adding the following to Nginx Dockerfule
```
ARG NGINX_LOCAL_INSTALL_SSL=${NGINX_LOCAL_INSTALL_SSL}
RUN if [ ${NGINX_LOCAL_INSTALL_SSL} = true ]; then \
RUN openssl genrsa -out "/etc/nginx/ssl/default.key" 2048 \
    && openssl req -new -key "/etc/nginx/ssl/default.key" -out "/etc/nginx/ssl/default.csr" -subj "/CN=default/O=default/C=NL" \
    && openssl x509 -req -days 365 -in "/etc/nginx/ssl/default.csr" -signkey "/etc/nginx/ssl/default.key" -out "/etc/nginx/ssl/default.crt" \
;fi
```
but I am still on the fence on this one.

**Comment by Shchepotin** [comment]
I think it final version:

- Added installation OpenSSL and autogeneration "default" certificate.
- Was create volume for certificates.
- Added SSL to sites config.

**Comment by ludo237** [comment]
This PR broke `startup.sh` on Windows 10 https://github.com/laradock/laradock/issues/1878

**Comment by Shchepotin** [comment]
Fixed it #1879

**Comment by andreladocruz** [comment]
@Shchepotin can you please help explaining how to enable Open SSL for development?

My laravel app responds to 2 different domains (domain1.test and domain2.test).

Do you have the "step by step" process to enable both with HTTPS?

**Comment by Shchepotin** [comment]
@andreladocruz 

- Uncomment this lines in your configs for domain1.test and domain2.test
https://github.com/laradock/laradock/blob/913abc54f9edde865a1d6de8a03b38098593ebfb/nginx/sites/laravel.conf.example#L7-L10
- Start/Restart containers.
- Open https://domain1.test

**Comment by andreladocruz** [comment]
@Shchepotin , thanks for the kind and detailed reply. =)

O got it working.

**Pull request merged**